contracts/modules/TransferManager/ManualApprovalTransferManager.sol

```diff
@@ -86,14 +99,47 @@ contract ManualApprovalTransferManager is ITransferManager {
     * @param _allowance is the approved amount of tokens
     * @param _expiryTime is the time until which the transfer is allowed
     */
-    function addManualApproval(address _from, address _to, uint256 _allowance, uint256 _expiryTime) public withPerm(TRANSFER_APPROVAL) {
+    function addManualApproval(address _from, address _to, uint256 _allowance, uint256 _expiryTime, bytes32 _description) public withPerm(TRANSFER_APPROVAL) {
         require(_to != address(0), "Invalid to address");
         require(_expiryTime > now, "Invalid expiry time");
         require(manualApprovals[_from][_to].allowance == 0, "Approval already exists");
-        manualApprovals[_from][_to] = ManualApproval(_allowance, _expiryTime);
+        manualApprovals[_from][_to] = ManualApproval(_from, _to, _allowance, _expiryTime, _description);
+        activeManualApprovals.push(ManualApproval(_from, _to, _allowance, _expiryTime, _description));
+
         emit AddManualApproval(_from, _to, _allowance, _expiryTime, msg.sender);
     }
 
+
+    /**
+    * @notice Adds mutiple manual approvals in batch
+    * @param _from is the address array from which transfers are approved
+    * @param _to is the address array to which transfers are approved
+    * @param _allowance is the array of approved amounts 
+    * @param _expiryTime is the array of the times until which eath transfer is allowed
+    * @param _description is the description array for these manual approvals
+    */
+    function addManualApprovalMulti(address[] _from, address[] _to, uint256[] _allowance, uint256[] _expiryTime, bytes32[] _description) public withPerm(TRANSFER_APPROVAL) {
+        require(_from.length == _to.length && _to.length == _allowance.length && _allowance.length == _expiryTime.length && _expiryTime.length == _description.length, "input array numbers not matching");
+        for(uint8 i=0; i<_from.length; i++){
+            addManualApproval(_from[i], _to[i], _allowance[i], _expiryTime[i], _description[i]);
```

Comment:
Create an internal `_addManualApproval()` function and reuse it here and in `addManualApproval` function to save gas.